src/Uno.Extensions.Reactive/Core/Internal/SourceContext.cs

```diff
@@ -89,28 +90,88 @@ public static void Set(object owner, SourceContext ctx)
 		_contexts.Add(owner, ctx);
 	}
 
+	/// <summary>
+	/// Creates a child context given a request source.
+	/// </summary>
+	/// <param name="requests">The request source to use for the new context.</param>
+	/// <returns>A new child SourceContext.</returns>
+	internal SourceContext CreateChild(IRequestSource requests)
+		=> new(this, requests: requests);
+	#endregion
+
+	private static long _nextRootId = 1;
+
 	private readonly bool _isNone;
 	private readonly CancellationTokenSource? _ct;
-	private Dictionary<object, IAsyncDisposable>? _states;
 
+	private readonly IStateStore? _localStates;
+	private readonly IRequestSource? _localRequests;
+
+	// Creates a root (or none) context
 	private SourceContext(bool isNone = false)
 	{
-		_isNone = isNone;
+		if (isNone)
+		{
+			_isNone = isNone;
 
-		if (!_isNone)
+			States = new NoneStateStore();
+			RequestSource = new NoneRequestSource();
+		}
+		else
 		{
-			_states = new();
-			_ct = new();
+			_ct = new CancellationTokenSource();
 
 			Token = _ct.Token;
+			RootId = (uint)Interlocked.Increment(ref _nextRootId);
+			States = _localStates = new StateStore(this);
+			RequestSource = _localRequests = new NoneRequestSource(); // Currently we do not support messages directly on the root, using None allows AsyncFeed to complete enumeration
 		}
 	}
 
+	// Creates a sub context
+	private SourceContext(SourceContext parent, IStateStore? states = null, IRequestSource? requests = null)
+	{
+		if (parent._isNone)
+		{
+			throw new InvalidOperationException("Cannot create a sub context from None context.");
```

Comment:
Debug only error, or should it be more actionable?

Reply:
https://github.com/unoplatform/uno.extensions/commit/550ddaf2156bfdcc0901347740dc90f011051a16 (Might push it only in next PR depending of the result of the current build)